pkg/etcdcli/health.go

```diff
@@ -52,10 +52,13 @@ func getMemberHealth(ctx context.Context, etcdMembers []*etcdserverpb.Member) me
 		const defaultTimeout = 30 * time.Second
 		resChan := make(chan healthCheck, 1)
 		go func() {
-			ctx, cancel := context.WithTimeout(ctx, defaultTimeout)
+			ctxTimeout, cancel := context.WithTimeout(ctx, defaultTimeout)
 			defer cancel()
 
-			resChan <- checkSingleMemberHealth(ctx, member)
+			resChan <- checkSingleMemberHealth(ctxTimeout, member)
+			// closing here to avoid late replies to panic on resChan,
+			// the result will be considered a timeout anyway
+			close(resChan)
```

Comment:
can we comment the issue here for the future ?

Reply:
it's in the git commit :+1: